src/util.cpp

```diff
@@ -244,4 +246,38 @@ Thread gettid()
 {
     return Thread(syscall(SYS_gettid));
 }
+
+int get_cgroup_mountpoint_fd(std::string cgroup)
+{
+    std::ifstream mtab("/proc/mounts");
+
+    // This regex does not work when cgroupfs is mounted on a path containing whitespaces
+    // But i think people that mount important Linux filesystems on paths containing
+    // whitespaces should not be let anywhere near lo2s anyways
+    // /proc/mounts format:
+    // device mountpoint fs_type options freq passno
+    std::regex cgroup_regex(R"((\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\S+))");
+    std::smatch cgroup_match;
+    std::string line;
+    while (std::getline(mtab, line))
+    {
+        if (std::regex_match(line, cgroup_match, cgroup_regex))
+        {
+            // For the ancient cgroupfs mount point we have to use the one with perf_event
+            // in the options
+            if (cgroup_match[3].str() != "cgroup" || cgroup_match[4].str().find("perf_event"))
```

Comment:
Should be more like `match[3] == "cgroup2" || (match[3] == cgroup && match[4] == perf_event)`